src/incubator/WheelPicker/index.tsx

```diff
@@ -2,55 +2,105 @@
 import View from '../../components/view';
+import Fader, {FaderPosition} from '../../components/fader';
 import {Constants} from '../../helpers';
-
+import useMiddleIndex from './helpers/useListMiddleIndex';
 import Item, {ItemProps} from './Item';
 
 const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
 
 export interface WheelPickerProps {
+  /**
+   * Data source for WheelPicker
+   */
   items?: ItemProps[];
+  /**
+   * Describe the height of each item in the WheelPicker
+   */
   itemHeight?: number;
-  itemTextStyle?: StyleProp<TextStyle>;
-  onChange: (item: ItemProps, index: number) => void;
+  /**
+   * TextStyle for the focused row
+   */
+  selectedTextColor?: string;
+  /**
+   * TextStyle for other, non-focused rows
+   */
+  unselectedTextStyle?: string;
+  /**
+   * Row text style
+   */
+  textStyle?: TextStyle;
+  /**
+   * Event, on active row change
+   */
+  onChange: (index: number, item?: ItemProps) => void;
 }
 
-const WheelPicker = ({items, itemHeight = 48, itemTextStyle}: WheelPickerProps) => {
-  const height = itemHeight * 4;
-  const scrollview = useRef<Animated.ScrollView>();
+const WheelPicker = ({items, itemHeight = 48, selectedTextColor, unselectedTextStyle, textStyle, onChange: onChangeEvent}: WheelPickerProps) => {
+  const height = itemHeight * 5;
+  const scrollView = useRef<Animated.ScrollView>();
   const [offset] = useValues([0], []);
   const onScroll = onScrollEvent({y: offset});
+  
+  const listSize = items?.length || 0;
+  const middleIndex = useMiddleIndex({itemHeight, listSize});
 
   const selectItem = useCallback(
     index => {
-      if (scrollview.current) {
-        scrollview.current.getNode().scrollTo({y: index * itemHeight, animated: true});
+      if (scrollView.current?.getNode()) {
+        //@ts-ignore for some reason scrollToOffset isn't recognized
+        scrollView.current.getNode().scrollToOffset({offset: index * itemHeight, animated: true});
       }
     },
     [itemHeight]
   );
 
-  const onChange = useCallback(() => {
-    // TODO: need to implement on change event that calc the current selected index
-  }, [itemHeight]);
-
+  const onChange = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const index = middleIndex(event.nativeEvent.contentOffset.y);
+    onChangeEvent(index, items?.[index]);
+  }, []);
+  
   const renderItem = useCallback(({item, index}) => {
     return (
       <Item
         index={index}
         itemHeight={itemHeight}
         offset={offset}
-        textStyle={itemTextStyle}
+        activeColor={selectedTextColor}
+        inactiveColor={unselectedTextStyle}
+        style={textStyle}
         {...item}
         onSelect={selectItem}
       />
     );
+  }, [itemHeight]);
+
+  const renderFader = useMemo(
+    () => (position: FaderPosition) => {
+      return <Fader visible position={position} size={60} />;
+    },
+    []
+  );
+
+  const renderSeparators = useMemo(() => {
```

Comment:
As discussed in last dev sync, I think it's more readable to name `useMemo` values as the object it returns 
```suggestion
  const separators = useMemo(() => {
```